examples/nlp/language_modeling/conf/megatron_llama_config.yaml

```diff
@@ -0,0 +1,219 @@
+name: megatron_llama
+restore_from_path: null # used when starting from a .nemo file
+
+trainer:
+  devices: 1
+  num_nodes: 1
+  accelerator: gpu
+  precision: 32
```

Comment:
@blahBlahhhJ @lhb8125 
is it possible to write weights in bf16 O2 at coversion? I'm not familiar with how HF handles weights for llama but the original model was trained with bf16 so would prefer if we can write the model directly in bf16 so we consume less space and can also easily use it directly

Reply:
no it converts it into O1 format and doesn't support O2 format. 

That being said, I convert the entire model to the target precision before saving it. so if precision is 16 or bf16, the storage it takes is one half of the full fp32 size. Most of the examples in Nemo can configure amp O2 at initialization so that's also not a problem.

Reply:
> no it converts it into O1 format and doesn't support O2 format.

whats the reason for this? since the original model was trained with bf16

I'm confused with what is happening with precision, since we're converting model from HF, what precision does HF keep the model weights in? are the model weights ever going to fp16/having lossy conversion?

Reply:
hf stores the model in fp16. When loading the model, it becomes fp32. and then I re-map the state dict keys into Nemo (still fp32), finally I cast the model to whatever precision set in the command and save it. (So I think technically fp16->fp32->fp16) doesn't lose info.

Reply:
Meta's original checkpoints are in bf16, HF's conversion to fp16 is dangerous...

Reply:
idk maybe it's for compatibility. some GPUs don't support bf16.